arch/mips/soc/danube/danube_irq.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2017 Imagination Technologies Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief danube interrupt management code
+ */
+
+#include <irq.h>
+#include <soc.h>
+
+void _arch_irq_enable(unsigned int irq)
+{
+	int key;
+
+	key = irq_lock();
+	mips_bissr(1 << (irq + 8));
```

Comment:
where is mips_bissr (and other related functions) actually implemented?
is it some kind of runtime library?
we would prefer zephyr to be stand-alone, if these are just functions to poke registers can you port them directly to zephyr?